uint/src/uint.rs

```diff
@@ -950,56 +1003,18 @@ macro_rules! construct_uint {
 			}
 		}
 
-		// specialization for u32
-		impl $crate::core_::ops::Mul<u32> for $name {
```

Comment:
That's a breaking API change. I think the reason why it wasn't implemented for all numbers was type interference. Now if you try to write `U256::from(2) * 2` it's going to complain, so you need to specify the correct type (brace for fixing the tests in dependent projects).
That said I think it's a good change, the previous behaviour was weird.

Reply:
it's generated via `impl_mul_for_primitive` now (also for other primitive types)

Reply:
Yeah, that's my point. Existing code will break cause there are now multiple (potentially conflicting) implementations of `Mul` if the type is not strictly specified (i.e. you just write `* 2` instead of `* 2_u8` or `* 2_u32`)

Reply:
> That's a breaking API change.

Indeed. We'll need a 0.7 release for sure.

Reply:
Ah, I see your point, it's inferred as i32 by default: https://play.rust-lang.org/?version=nightly&mode=release&edition=2018&gist=f3e5e51ddb3d1fa2ef073ae0085d6199